libr/core/cmd_open.c

```diff
@@ -678,10 +680,21 @@ static void cmd_open_map(RCore *core, const char *input) {
 				}
 			}
 		} else {
-			if (!(s = strdup (&input[2]))) {
+			bool use_id = false;
+			if (input[2] == 'i') {
+				use_id = true;
+			}
+
```

Comment:
Use ternary operator on strdup argument not at assignment level

Reply:
you mean:

<pre>
s = (use_id) ? strdup (&input[3]) : strdup (&input[2]);
</pre>